include/pcmki/pcmki_transition.h

```diff
@@ -28,19 +28,39 @@ typedef enum {
 typedef struct te_timer_s crm_action_timer_t;
 typedef struct crm_graph_s crm_graph_t;
 
+enum pcmki__synapse_flags {
+    pcmki__synapse_ready       = (1 << 0),
+    pcmki__synapse_failed      = (1 << 1),
+    pcmki__synapse_executed    = (1 << 2),
+    pcmki__synapse_confirmed   = (1 << 3),
+};
+
 typedef struct synapse_s {
     int id;
     int priority;
 
-    gboolean ready;
-    gboolean failed;
-    gboolean executed;
-    gboolean confirmed;
+    uint32_t flags; // Group of pcmki__synapse_flags
 
     GList *actions;           /* crm_action_t* */
     GList *inputs;            /* crm_action_t* */
 } synapse_t;
 
+const char *pcmk__synapse_name(synapse_t *c);
+
+#define pcmk__set_synapse_flags(synapse, flags_to_set) do {             \
+        (synapse)->flags = pcmk__set_flags_as(__func__, __LINE__,       \
+            LOG_TRACE,                                                  \
+            "Synapse", pcmk__synapse_name(synapse),                       \
```

Comment:
There's no pcmk__synapse_name() defined -- I'm guessing you meant to expose synapse_state_str() and rename it?